File: programming 1/lab 6/src/main/java/client/managers/ResponseManager.java
package client.managers;

import client.network.Client;
import common.communication.Response;
import common.utils.ObjectBuilder;

/**
 * Менеджер обработки ответов от сервера.
 */
public class ResponseManager {
    private final Client client;

    public ResponseManager(Client client) {
        this.client = client;
    }

    /**
     * Обрабатывает ответ от сервера.
     * @param response ответ сервера.
     */
    public void handle(Response response) {
        switch (response.getStatus()) {
            case SUCCESS, INTERMEDIATE_SUCCESS -> System.out.println(response.getDescr());
            case ERROR, INTERMEDIATE_ERROR, HISTORY_OVERFLOW -> System.err.println(response.getDescr());
            case ASK_OBJECT -> {
                client.sendRequest(ObjectBuilder.create());
            }
            case EXIT -> {
                System.out.println(response.getDescr());
                System.exit(0);
            }
            default -> System.out.println("не учла папапаараша");
        }
    }
}

File: programming 1/lab 6 kotlin/kotlin/lab 6/src/main/java/client/managers/ResponseManager.kt
package client.managers

import client.network.Client
import common.communication.Response
import common.communication.ResponseStatus
import common.utils.ObjectBuilder
import kotlin.system.exitProcess

/**
 * Менеджер обработки ответов от сервера.
 */
class ResponseManager(private val client: Client) {
    /**
     * Обрабатывает ответ от сервера.
     * @param response ответ сервера.
     */
    fun handle(response: Response) {
        when (response.status) {
            ResponseStatus.SUCCESS, ResponseStatus.INTERMEDIATE_SUCCESS -> println(response.descr)
            ResponseStatus.ERROR, ResponseStatus.INTERMEDIATE_ERROR, ResponseStatus.HISTORY_OVERFLOW -> System.err.println(
                response.descr
            )

            ResponseStatus.ASK_OBJECT -> {
                client.sendRequest(ObjectBuilder.create())
            }

            ResponseStatus.EXIT -> {
                println(response.descr)
                exitProcess(0)
            }
//            else -> println("не учла папапаараша")
        }
    }
}
